src/package-request.js

```diff
@@ -369,16 +372,34 @@ export default class PackageRequest {
           throw new MessageError(reporter.lang('lockfileOutdated'));
         }
 
-        const {name, version: current} = locked;
+        const {name, version: current, resolved} = locked;
         let latest = '';
         let wanted = '';
         let url = '';
+        let isOldHash = true;
 
         const normalized = normalizePattern(pattern);
 
-        if (getExoticResolver(pattern) || getExoticResolver(normalized.range)) {
+        const exoticResolver = getExoticResolver(pattern) || getExoticResolver(normalized.range);
+        if (exoticResolver) {
           latest = wanted = 'exotic';
           url = normalized.range;
+
+          // check whether the exotic version has a new commit on git.
+          if (exoticResolver === GitResolver) {
```

Comment:
Flow gets mad at this because it thinks `resolved` can be null/undefined, but the function `versionUtil.explodeHashedUrl(resolved)` does not accept null.

You can fix this by adding a check to the `if`:

```
if (exoticResolver === GitResolver && resolved) {
```